### PR TITLE
fix: Ensure all widgets can be re-targeted

### PR DESCRIPTION
**Pull request by GordonSmith**
Added retarget to core unit test.

Signed-off-by: Gordon Smith <gordonjsmith@gmail.com>

## Checklist:
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit message includes a "fixes" reference if appropriate.
  - [x] The commit is signed.
- [x] The change has been fully tested:
  - [x] I have viewed all related gallery items
  - [ ] I have viewed all related dermatology items
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised new issues to address them separately

### CODE REVIEW

File: packages/chart/src/Summary.ts [hunk 1]
```diff
@@ -101,7 +101,7 @@ export class Summary extends HTMLWidget {
     }
 
     update(_domNode, element) {
-        HTMLWidget.prototype.update.apply(this, arguments);
+        super.update(_domNode, element);
```
Comment (jbrundage) [inline packages/chart/src/Summary.ts]: You missed an underscore here

File: packages/common/src/Entity.ts [hunk 2]
```diff
@@ -128,6 +128,13 @@ export class Entity extends SVGWidget {
             })
             .remove();
     }
+    exit(dn, element) {
```
Comment (jbrundage) [inline packages/common/src/Entity.ts]: domNode

File: packages/common/src/Surface.ts [hunk 3]
```diff
@@ -59,7 +59,7 @@ export class Surface extends SVGWidget {
     }
 
     enter(_domNode, _element) {
-        SVGWidget.prototype.enter.apply(this, arguments);
+        super.enter(_domNode, _element);
```
Comment (jbrundage) [inline packages/common/src/Surface.ts]: Underscores

File: packages/common/src/Widget.ts [hunk 4]
```diff
@@ -530,6 +530,26 @@ export abstract class Widget extends PropertyExt {
         return this._isRootNode && this._placeholderElement.node().offsetParent === null;
     }
 
+    protected ppWidgets(): Widget[] {
```
Comment (jbrundage) [inline packages/common/src/Widget.ts]: This method could use a more descriptive name

File: packages/common/src/Surface.ts (hunk 3 above)
Comment (GordonSmith) [reply packages/common/src/Surface.ts]: This is correct here.

**Comment by GordonSmith** [comment]
@jbrundage changes made as requested (except one).

**Comment by jbrundage** [comment]
@GordonSmith couldn't we avoid a lot of that `target(null)` code by doing something like this at the `Widget.ts` level?
```javascript
this._publishedProperties.forEach(pp=>{
    if(pp.type === "widgetArray"){
        this[pp.label]().forEach(widget=>{
            if(widget.target){
                widget.target(null);
            }
        });
    }
});
```

File: packages/tree/src/Treemap.ts [hunk 5]
```diff
@@ -97,16 +97,16 @@ export class Treemap extends HTMLWidget {
...
+        super.enter(domNode, element);
         this._d3Treemap = d3Treemap();
 
         this._elementDIV = element.append("div");
         this._selection.widgetElement(this._elementDIV);
     }
 
     update(_domNode, _element) {
-        HTMLWidget.prototype.update.apply(this, arguments);
+        super.update(_domNode, _element);
```
Comment (jbrundage) [inline packages/tree/src/Treemap.ts]: Underscores

**Comment by GordonSmith** [comment]
> @GordonSmith couldn't we avoid a lot of that `target(null)` code by doing something like this at the `Widget.ts` level?
> 
> ```js
> this._publishedProperties.forEach(pp=>{
>     if(pp.type === "widgetArray"){
>         this[pp.label]().forEach(widget=>{
>             if(widget.target){
>                 widget.target(null);
>             }
>         });
>     }
> });
> ```

Yes and you reviewd that code:  https://github.com/hpcc-systems/Visualization/pull/3131/files#diff-fa7e5abc740bbfa6656f7e3f7ba0f128R664

**Pull request merged**